tools/packaging/common/envoy_bootstrap_v2.json

```diff
@@ -239,182 +239,101 @@
-              "port_value": {{ .config.ProxyAdminPort }}
-            }
-          }
-        ]
+        "load_assignment": {
+          "cluster_name": "prometheus_stats",
+          "endpoints": [{
+            "lb_endpoints": [{
+              "endpoint": {
+                "address":{
+                  "socket_address": {
+                    "protocol": "TCP",
+                    "address": "{{ .localhost }}",
+                    "port_value": {{ .config.ProxyAdminPort }}
+                  }
+                }
+              }
+            }]
+          }]
+        }
       },
-      {{- if eq .sds_uds_path "unix:/etc/istio/proxy/SDS" }}
       {
         "name": "sds-grpc",
         "type": "STATIC",
         "http2_protocol_options": {},
         "connect_timeout": "{{ .connect_timeout }}",
         "lb_policy": "ROUND_ROBIN",
-        "hosts": [{
-          "pipe": {
-            "path": "/etc/istio/proxy/SDS"
-           }
-        }]
+        "load_assignment": {
+          "cluster_name": "sds-grpc",
+          "endpoints": [{
+            "lb_endpoints": [{
+              "endpoint": {
+                "address":{
+                  "pipe": {
+                    "path": "/etc/istio/proxy/SDS"
+                  }
+                }
+              }
+            }]
+          }]
+        }
       },
-      {{- end }}
       {
         "name": "xds-grpc",
         "type": "STRICT_DNS",
         "respect_dns_ttl": true,
         "dns_lookup_family": "{{ .dns_lookup_family }}",
         "connect_timeout": "{{ .connect_timeout }}",
         "lb_policy": "ROUND_ROBIN",
-        {{ if eq .config.ControlPlaneAuthPolicy 1 }}
-        {{ if .sds_uds_path }}
-        "tls_context": {
-          "common_tls_context": {
-            "alpn_protocols": [
-              "h2"
-            ],
-            {{ if eq .sds_uds_path "unix:/etc/istio/proxy/SDS" }}
-             "tls_certificate_sds_secret_configs":[
-              {
-                "name":"default",
-                "sds_config":{
-                  "api_config_source":{
-                    "api_type":"GRPC",
-                    "grpc_services":[
-                      {
-                        "envoy_grpc":{
-                          "cluster_name": "sds-grpc"
+        {{- if eq .config.ControlPlaneAuthPolicy 1 }}
+        "transport_socket": {
+          "name": "envoy.transport_sockets.tls",
+          "typed_config": {
+            "@type": "type.googleapis.com/envoy.api.v2.auth.UpstreamTlsContext",
+            "common_tls_context": {
+              "alpn_protocols": [
+                "h2"
+              ],
+              "tls_certificate_sds_secret_configs": [
+                {
+                  "name": "default",
+                  "sds_config": {
+                    "api_config_source": {
+                      "api_type": "GRPC",
+                      "grpc_services": [
+                        {
+                          "envoy_grpc": { "cluster_name": "sds-grpc" }
```

Comment:
Need google_grpc for STS?

Reply:
this is SDS, I'm not sure how STS plays in here, is it in the same path? note this is not changing in this pr but was changed a couple weeks ago.

For sds, we don't need Google grpc because SDS server is now on the pilot-agent, which already had access to the same credentials. I would imagine STS is the same? Previously we needed this to pass the credentials from envoy to nodeagent which was in another pod, but that is no longer required.

That's my understanding of this at least

Reply:
Ok, but it looks like a functional change. Previously, there was some sort of authz on SDS server using STS. I don't really have any context on this but thought it's worth pointing out that this is not just refactoring to v3 xDS.

Reply:
I am not changing that code though. There's 3 paths - file mount, legacy nodeagent SDS using Google grpc, and the istio-agent SDS using envoy grpc. I removed the legacy nodeagent one, but didn't change the behavior of anything.

I could probably break out the change to remove the legacy SDS stuff though to make this pr smaller scoped